viewing_party/party.py

```diff
@@ -1,23 +1,155 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    new_watchlist = []
+
+    for i in range(len(watchlist)):

```

Comment:
This is an interesting approach! A couple of suggestions - you don't really need the index here so it would be a bit tidier to just iterate over the movies in the watch list directly `for movie in watchlist:` so you can then use `movie["title"]` and pass `movie` to `add_to_watched`. 

Just a quick word on modifying a collection while iterating over it (here, it's the watchlist). It's a little dangerous to do, in general, although in this case, it ends up being OK, since we really only expect a single copy of a movie to be in the watchlist at a time. However, if we were iterating over a list, and there was the possibility of multiple adjacent items being in the list that we want to remove, we could end up skipping over items as the list items get shifted around underneath the iteration.

If we know for certain that we only need to remove one thing, we could break out of the loop as soon as we've removed the one thing (using break). Alternatively, we could find the movie record, and then wait to perform the remove until after completing that loop.

Again, nothing goes wrong in the current situation, but in general, we should be wary of modifying a collection while we are iterating through it.

In this case too, you'd have to modify the approach so that the appending and removing happen in the same conditional block.